auth/src/api/handlers.rs

```diff
@@ -68,6 +65,20 @@ pub(crate) async fn update_user_tier(
     Ok(())
 }
 
+// #[debug_handler]
+#[instrument(skip(claim, user_manager), fields(account_name = claim.sub, account_tier = %claim.tier))]
+pub(crate) async fn add_subscription_items(
+    Extension(claim): Extension<Claim>,
+    State(user_manager): State<UserManagerState>,
+    NewSubscriptionItemExtractor(update_subscription_items): NewSubscriptionItemExtractor,
+) -> Result<(), Error> {
+    user_manager
+        .add_subscription_items(AccountName::from(claim.sub), update_subscription_items)
+        .await?;
+
+    Ok(())
+}
```

Comment:
Don't we check anything about the claim?

Reply:
Hmm, what should we check? We check in the jwt auth layer that it isn't expired.

Reply:
I was thinking about checking if the claim has a scope for updating the subscription. Are we checking it downstream?

Reply:
We technically are both upstream and downstream, this endpoint is currently only called when rds is provisioned in the provisioner, which is only done if the claim has access to RDS. Before we send the update request to stripe downstream, we also check that the account tier is still pro, which means they can provision RDS. I can add a check at the handler level either way, though, since the upstream and downstream may change in the future.

Reply:
I added a scopedlayer now, restricted to "ResourcesWrite". We can consider adding a new scope for updating subscription, but I'm not sure it's necessary yet.

Adding the scopedlayer fixed the unexpected 500 from missing token: https://github.com/shuttle-hq/shuttle/pull/1478#discussion_r1431223035